### PR TITLE
[Dashboard] Fix falsy logic when updating Royalties and PlatformFees

### PR DESCRIPTION
**Pull request by kien-ngo**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR focuses on refining the validation logic for platform fees and royalties by removing default values and error handling related to `platform_fee_basis_points` and `platform_fee_recipient` in `platform-fees.tsx` and `royalties.tsx`.

### Detailed summary
- Removed default values for `platform_fee_basis_points` and `platform_fee_recipient` in `platform-fees.tsx`.
- Removed error handling for invalid `platform_fee_basis_points` and `platform_fee_recipient` in `platform-fees.tsx`.
- Removed default values for `seller_fee_basis_points` and `fee_recipient` in `royalties.tsx`.
- Removed error handling for invalid `seller_fee_basis_points` and `fee_recipient` in `royalties.tsx`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Review by MananTank — APPROVED** [review]

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/settings/components/platform-fees.tsx [hunk 1]
```diff
@@ -84,11 +84,6 @@ export const SettingsPlatformFees = ({
             action: "set-platform-fees",
             label: "attempt",
           });
-          // In the v4 hook we defaulted recipient to address_zero and bps to `0`
-          // but let's actually throw an error here for better transparency
-          if (!data.platform_fee_basis_points) {
-            return toast.error("Please enter valid basis points.");
-          }
           if (!data.platform_fee_recipient) {
```
Comment (MananTank) [inline apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/settings/components/platform-fees.tsx:87]: remove this as well

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/settings/components/royalties.tsx [hunk 2]
```diff
@@ -97,11 +97,6 @@ export const SettingsRoyalties = ({
             action: "set-royalty",
             label: "attempt",
           });
-          // In the v4 hook we defaulted recipient to address_zero and bps to `0`
-          // but let's actually throw an error here for better transparency
-          if (!d.seller_fee_basis_points) {
-            return toast.error("Please enter valid basis points.");
-          }
           if (!d.fee_recipient) {
```
Comment (MananTank) [inline apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/settings/components/royalties.tsx:100]: remove this as well

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/settings/components/platform-fees.tsx [hunk 3]
```diff
@@ -36,7 +36,7 @@ const CommonPlatformFeeSchema = z.object({
   /**
    * platform fee basis points
    */
-  platform_fee_basis_points: BasisPointsSchema.default(0),
+  platform_fee_basis_points: BasisPointsSchema,
   /**
    * platform fee recipient address
    */
```
Comment (MananTank) [inline apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/settings/components/platform-fees.tsx:42]: <i class='graphite__hidden'>[Re: line +43]<br/><br/></i>

remove default from here as well<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5342?utm_source=unchanged-line-comment">Graphite</a>.</span>

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/settings/components/royalties.tsx [hunk 4]
```diff
@@ -46,7 +46,7 @@ const CommonRoyaltySchema = z.object({
    * @internal
    * @remarks used by OpenSea "seller_fee_basis_points"
    */
-  seller_fee_basis_points: BasisPointsSchema.default(0),
+  seller_fee_basis_points: BasisPointsSchema,
 
   /**
    * The address of the royalty recipient. All royalties will be sent
```
Comment (MananTank) [inline apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/settings/components/royalties.tsx:52]: <i class='graphite__hidden'>[Re: line +57]<br/><br/></i>

remove default from here as well<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5342?utm_source=unchanged-line-comment">Graphite</a>.</span>

**Review by MananTank — CHANGES_REQUESTED** [review]
please check comments

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/settings/components/platform-fees.tsx (hunk 1 above)
Comment (kien-ngo) [reply apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/settings/components/platform-fees.tsx]: removed

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/settings/components/royalties.tsx (hunk 2 above)
Comment (kien-ngo) [reply apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/settings/components/royalties.tsx]: removed\

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/settings/components/royalties.tsx (hunk 4 above)
Comment (kien-ngo) [reply apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/settings/components/royalties.tsx:51]: removed

**Review by MananTank — APPROVED** [review]

**Pull request merged**